### PR TITLE
feat: added Linux Mint Support

### LINKED ISSUE #516: [FEAT] Linux Mint support
**Is your feature request related to a problem? Please describe.**
Because of several fixed configuration parameters, Linux Mint could not be used as target distribution.

**Describe the solution you'd like**
Two options for a solution:
1. Add necessary additional values to vars (A merge request for this case was provided in #515)
2. Move `tailscale_apt_deb` and `tailscale_apt_signkey` from vars to defaults (This would induce changes to dependent variables as well, e.g. the codename for the apt repository)

**Describe alternatives you've considered**
Alternatively, the user can download the .deb file via a pre-task and install a custom .deb file.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by fkonradmain**
Added Linux Mint support.

Following Problems induced changes:

1. the codename for `name: Debian | Add Tailscale Deb` would point to `wilma` but it should be `noble` (`viriginia` -> `jammy`, etc. respectively)
   Solution:
     - introduction of a mapping dict `tailscale_debian_release`, that replaces sub distro codenames to parent distro code names.
     - added Value for `codename` when calling `ansible.builtin.apt_repository`, but only if the codename has to be overridden

2. Linux Mint was not among the list of supported operating systems
   Solution: added entries for Linux Mint at `tailscale_debian_family_distros`
  

3. URLs would include `linux mint` or `wilma` instead of `ubuntu` and `noble` (e.g. `https://pkgs.tailscale.com/stable/linux mint/wilma.noarmor.gpg` instead of `https://pkgs.tailscale.com/stable/ubuntu/noble.noarmor.gpg`)
   Solution:
     - also added entries for Linux mint at `tailscale_debian_distro`
     - added functionality to actually look up table values from `tailscale_debian_distro` with automatic fallback to `ansible_distribution`

Currently supported:
- Linux Mint 22

Tested on:
- Linux Mint 22
- Debian 12

Closes #516

### CODE REVIEW

**Review by artis3n — CHANGES_REQUESTED** [review]
Thank you for submitting this PR!

In order to add a new supported distro, the following additional steps need to be added:

1. Update Galaxy metadata (https://github.com/artis3n/ansible-role-tailscale/blob/main/meta/main.yml#L11)
2. Build a testable Molecule container for the distro (e.g. https://github.com/geerlingguy/docker-ubuntu2404-ansible and https://github.com/artis3n/docker-rhel9-ansible)
3. Ensure the test suite runs successfully on the container (https://github.com/artis3n/ansible-role-tailscale/blob/main/Makefile#L32) (`MOLECULE_DISTRO=your-distro-here make test-all`)
4. Add the container to the CI suite (https://github.com/artis3n/ansible-role-tailscale/blob/main/.github/workflows/pull_request_target.yml#L26)

File: vars/main.yml [hunk 1]
```diff
@@ -36,12 +37,20 @@ tailscale_debian_distro:
...
   debian: debian
   osmc: debian
+  "linux mint": ubuntu
+
+tailscale_debian_release:
+  vanessa: jammy
+  vera: jammy
+  victoria: jammy
+  virginia: jammy
+  wilma: noble
```
[the remark this answers is not in the document]
Comment (fkonradmain) [reply vars/main.yml]: Resolved this by automatically retrieving `UBUNTU_CODENAME` on Ubuntu based distros, while still using `ansible_distribution_release` on debian based distros.

**Comment by artis3n** [comment]
Ah, TIL `platforms` isn't respected by Galaxy anymore anyway
https://forum.ansible.com/t/role-meta-data-platform-key-broken/4651/9

**Comment by artis3n** [comment]
Looks like `apt_codename` needs to be handled in the uninstall tasks: https://github.com/artis3n/ansible-role-tailscale/actions/runs/11873194056/job/33087762830?pr=515#step:6:277

**Comment by fkonradmain** [comment]
> Looks like `apt_codename` needs to be handled in the uninstall tasks: https://github.com/artis3n/ansible-role-tailscale/actions/runs/11873194056/job/33087762830?pr=515#step:6:277

I will take care of that.

File: .github/workflows/pull_request_target.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@ jobs:
           - image: ghcr.io/artis3n/docker-rhel8-ansible:latest
           - image: ghcr.io/artis3n/docker-oraclelinux8-ansible:latest
           - image: geerlingguy/docker-ubuntu2404-ansible:latest
+          - image: ghcr.io/fkonradmain/docker-linuxmint22-ansible:latest
```
Comment (fkonradmain) [inline .github/workflows/pull_request_target.yml:34]: @artis3n I don't see this pop up in the test suite. Does this need additional approval from your side as owner or did I miss anything?

**Comment by artis3n** [comment]
Why is this image 4+GB? The other ubuntu-based ones are much smaller 🤔

File: .github/workflows/pull_request_target.yml (hunk 2 above)
Comment (artis3n) [reply .github/workflows/pull_request_target.yml:34]: `pull_request_target` works by running the version of this file on the _main_ branch against the code we check out. Needed to safely use my tailscale key to run the CI. Side effect is any changes to this file won't reflect in tests until it is merged.

To test this container, I run `MOLECULE_DISTRO="ghcr.io/fkonradmain/docker-linuxmint22-ansible:latest" make test` locally (or the other test commands in the Makefile)

File: tasks/debian/common.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    awk -F= '/^UBUNTU_CODENAME/ {print $2}' {{ ansible_distribution_file_path }}
+  register: ubuntu_codename
+  when: apt_id_like.stdout == '"ubuntu debian"'
+  changed_when: false
+
+- name: Debian | Set codename for apt
+  ansible.builtin.set_fact:
+    apt_codename: "{{ ubuntu_codename.stdout | default(ansible_distribution_release) }}"
+
+- name: Debian | Apt Dependencies
```
Comment (artis3n) [inline tasks/debian/common.yml:21]: Why include this here? There is also the legacy apt dependency task.

Perhaps call this file `tasks/debian/distro_codename.yml` and just include the first 3 tasks.

File: tasks/debian/install.yml [hunk 4]
```diff
@@ -1,10 +1,7 @@
 ---
-- name: Debian | Apt Dependencies
-  become: true
-  ansible.builtin.apt:
-    name: "{{ tailscale_apt_dependencies }}"
-    cache_valid_time: 3600
-    state: present
+- name: Debian | Include common dependencies for install and uninstall
```
Comment (artis3n) [inline tasks/debian/install.yml:2]: If you make the changes in `common.yml`, then rename this task

File: tasks/debian/uninstall.yml [hunk 5]
```diff
@@ -1,10 +1,7 @@
 ---
-- name: Debian | Apt Dependencies
-  become: true
-  ansible.builtin.apt:
-    name: "{{ tailscale_apt_dependencies }}"
-    cache_valid_time: 3600
-    state: present
```
Comment (artis3n) [inline tasks/debian/uninstall.yml:7]: 🤔 I am not sure why I'm installing apt dependencies in the uninstall task to begin with. Need to look into that.

File: tasks/debian/apt-codename.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+# ID_LIKE has the following values, depending on the operating system
+# | OS               | ID_LIKE                |
+# | Debian           | -- does not exist --   |
+# | Raspberry Pi OS  | -- does not exist --   |
+# | Proxmox          | -- does not exist --   |
+# | Devuan           | "debian"               |
+# | Raspbian         | "debian"               |
+# | Ubuntu           | "debian"               |
+# | Pop!OS           | "ubuntu debian"        |
+# | Linux Mint       | "ubuntu debian"        |
```
Comment (artis3n) [inline tasks/debian/apt-codename.yml:12]: Great comment block, thank you!

File: tasks/debian/apt-codename.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+- name: Debian | Retrieve Apt ID_LIKE
+  check_mode: false
+  ansible.builtin.command: >-
+    awk -F= '/^ID_LIKE/ {print $2}' {{ ansible_distribution_file_path }}
+  register: apt_id_like
+  changed_when: false
+
+# Ubuntu and its derivatives all have a value called "UBUNTU_CODENAME" set in their "/etc/os-release"
+# This value points to the actual ubuntu Version t
```
Comment (artis3n) [inline tasks/debian/apt-codename.yml:21]: Small typo at the end of this!

**Review by artis3n — APPROVED** [review]
Thanks for making these changes!

File: tasks/debian/apt-codename.yml (hunk 7 above)
Comment (artis3n) [reply tasks/debian/apt-codename.yml:21]: @fkonradmain if you could just adjust this comment!

**Review by artis3n — APPROVED** [review]

File: tasks/debian/apt-codename.yml (hunk 7 above)
Comment (artis3n) [reply tasks/debian/apt-codename.yml]: Yeah something changed in the rockylinux distros, I don't think it is related to this PR. I need to go investigate it.

**Pull request merged**